src/HasTranslations.php

```diff
@@ -71,19 +71,20 @@ public function getTranslationWithoutFallback(string $key, string $locale): mixe
         return $this->getTranslation($key, $locale, false);
     }
 
-    public function getTranslations(string $key = null): array
+    public function getTranslations(string $key = null, array $locales = null): array
     {
         if ($key !== null) {
             $this->guardAgainstNonTranslatableAttribute($key);
 
             return array_filter(
                 json_decode($this->getAttributes()[$key] ?? '' ?: '{}', true) ?: [],
-                fn ($value) => $value !== null && $value !== ''
+                fn ($value, $locale) => $value !== null && $value !== '' && ($locales === null || in_array($locale, $locales ?? [])),
```

Comment:
To me, this is unreadable. Could you replace the conditional with a well named function that uses early returns?